setup.py

```diff
@@ -23,85 +24,44 @@ def get_tag(self):
         return python, abi, plat
 
 
-def abs_machine():
-    machine = platform.machine()
-
-    pointer_bits = struct.calcsize("P") * 8
-    if pointer_bits not in (32, 64):
-        raise Exception("Unsupported pointer size")
-
-    is_64 = pointer_bits == 64
-
-    # x86 based archs
-    if machine in ('AMD64', 'x86_64', 'i686'):
-        return "x86_64" if is_64 else "i686"
-    # arm based archs
-    elif machine in ('aarch64', 'arm64', 'armv6l', 'armv7l'):
-        return "aarch64" if is_64 else "arm"
-    else:
-        raise Exception("Unsupported processor")
+def detect_arch():
```

Comment:
again we have duplicated detection !?

Reply:
Sorry for the misleading before, it turns out that we can download_so in build.py :)